velox/docs/monthly-updates/january-2023.rst

```diff
@@ -0,0 +1,76 @@
+********************
+January 2023 Update
+********************
+
+Core Library
+============
+
+* Add generic writer which supports generic outputs in the simple function interface and thus unblocks authoring a large set of functions that operate on any type such as :func:`subscript`, :func:`array_concat`, :func:`array_intersect`.
+* Reject null aware right semi project joins and add support for null-aware left semi join project with filter.
+* Fix memory pool: to always return a valid buffer, memory usage count leak with async data cache failures.
+* Make memory usage tracker thread safe.
+
+
+Presto Functions
+================
+
+* Add :func:`array_frequency`, :func:`json_parse`, :func:`shuffle`, :func:`sha1`, :func:`spooky_hash_v2_32`, :func:`spooky_hash_v2_64`, :func:`repeat`, :func:`week`, :func:`week_of_year`, :func:`json_format` scalar functions.
+* Fix error handling in :func:`least`, :func:`greatest`, :func:`like`, :func:`array_sum`, :func:`array_position`, :func:`regex_extract`, :func:`regex_like`, :func:`regex_extract_all` scalar functions.
+* Fix bugs in :func:`array_except`, :func:`intersect`, :func:`date_parse`, :func:`bitwise_left_shift`, :func:`bitwise_right_shift` scalar functions.
+* Fix crash in :func:`approx_distinct` when passing in minimum error bound.
+
+
+Spark Functions
+===============
+
+* Add support for DATE inputs to least and greatest functions.
+* Fix year function to match Spark's semantics.
+
+Hive Connector
+==============
+
+* Fix crashes when reading from an S3 file system.
+* Add support for MAP columns in Parquet reader.
+* Fix is-null and is-not-null filter pushdown for ARRAY and MAP columns.
+* Fix unnecessary copy in the native Parquet reader.
+
+
+Substrait
+=========
+
+* Update Substrait to 0.22.0.
+
+
+Performance and Correctness
+===========================
+
+* Fix multiple bugs in expression evaluation with TRY.
+* `applyFunctionWithPeeling` should flatten single argument functions input always.
+* Modify expression fuzzer to randomly choose functions from a list of eligible function names instead of from a list of function signatures.
```

Comment:
to add more context:
"Revamp function selection mechanism in expression fuzzer. This change will promote more even distribution of test coverage across functions, as the selection of functions will now be based on a list of eligible function names, rather than being based on function signatures, which could lead to an uneven distribution based on factors such as the presence of a templated return type or the number of signatures a function has."

Reply:
@bikramSingh91 This is nice.

@kgpai Let's update.